cartographer/transform/rigid_transform_test.cc

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright 2017 The Cartographer Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "cartographer/transform/rigid_transform.h"
+#include "cartographer/transform/rigid_transform_test_helpers.h"
+
+#include <random>
+
+#include "gtest/gtest.h"
+
+namespace cartographer {
+namespace transform {
+namespace {
+
+template <typename T>
+class RigidTransformTest : public ::testing::Test {
+ protected:
+  RigidTransformTest() {}
+
+  T eps();
+
+  Rigid2<T> GetRandomRigid2() {
+    const T x = T(0.7) * distribution_(prng_);
+    const T y = T(0.7) * distribution_(prng_);
+    const T theta = T(0.2) * distribution_(prng_);
+    return transform::Rigid2<T>(typename Rigid2<T>::Vector(x, y), theta);
+  }
+
+  Rigid3<T> GetRandomRigid3() {
+    const T x = T(0.7) * distribution_(prng_);
+    const T y = T(0.7) * distribution_(prng_);
+    const T z = T(0.7) * distribution_(prng_);
+    const T ax = T(0.7) * distribution_(prng_);
+    const T ay = T(0.7) * distribution_(prng_);
+    const T az = T(0.7) * distribution_(prng_);
+    const T theta = T(0.2) * distribution_(prng_);
```

Comment:
You could use `AngleAxisVectorToRotationQuaternion()`.

Reply:
Hmm, I don't understand how this would make things more concise.

Reply:
It only needs the `(ax, ay, az)` vector. No need for this line then anymore.

Reply:
Done